dataset/src/test/scala/frameless/SelectTests.scala

```diff
@@ -18,9 +18,10 @@ class SelectTests extends TypedDatasetSuite {
       val A = dataset.col[A]('a)
 
       val dataset2 = dataset.select(A).collect().run().toVector
+      val symDataset2 = dataset.select(functions.col('a)).collect().run().toVector
```

Comment:
Interesting, this works because inference fixes `T` from the expected type of select. But does it scale to complex expressions?

Reply:
Good point. Yes, this will probably only work when used directly in the args to select. Not sure if it’ll work with selectMany, that needs to be tested. I plan on adding a similar assertion to all of the tests here.

An idea to deal with situations in which you have to specify T is to use https://tpolecat.github.io/2015/07/30/infer.html.

Reply:
Hmm.. but if `T` is `Tuple3[Int, String, Double]` it's not gonna look pretty :P

Reply:
True. I have another idea - will push in a few minutes.